lib/http.c

```diff
@@ -925,7 +925,7 @@ CURLcode Curl_http_input_auth(struct connectdata *conn, bool proxy,
             }
           }
           else
-            if(checkprefix("Bearer", auth)) {
+            if(conn->oauth_bearer && checkprefix("Bearer", auth)) {
```

Comment:
This is probably the wrong spot to do this. I think `pickoneauth()` is the correct place. Let me test this.

Reply:
I just force-pushed an update.

The reason: I was not comfortable meddling with the `avail` bit-field, as the server *did* report it as available authentication method, and users of libcurl will see that information.

Instead, the patch now really only verifies at *`pickoneauth()`* time that we pick the Bearer method only when we have a Bearer token.

Note: for the Basic authentication method, IIRC it is totally legitimate to pick it without valid credentials, and then kick that result back to the caller, so that credentials can be provided interactively. But for the Bearer authentication, I think it makes no sense to ask for a token interactively, like, ever: you either have it, or you don't. And if you don't have a token, let's not try to authenticate with it ;-)

Reply:
Oh yes, and of course I verified that the force-pushed patch *also* fixes the symptom on my side.